src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from "azure-arm-website";
+import { DotenvParseOutput } from "dotenv";
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { AppSettingsTreeItem, SiteClient } from "vscode-azureappservice";
+import { envFileName } from "../../constants";
+import { ext } from "../../extensionVariables";
+import * as workspaceUtil from '../../utils/workspace';
+import { confirmOverwriteSettings } from "./confirmOverwriteSettings";
+import { getLocalEnvironmentVariables } from './getLocalEnvironmentVariables';
+
+export async function downloadAppSettings(node?: AppSettingsTreeItem): Promise<void> {
+    if (!node) {
+        node = <AppSettingsTreeItem>await ext.tree.showTreeItemPicker(AppSettingsTreeItem.contextValue);
+    }
+
+    const client: SiteClient = node.root.client;
+
+    const message: string = 'Select the destination file for your downloaded settings.';
+    const envVarPath: string = await workspaceUtil.selectWorkspaceFile(ext.ui, message, () => envFileName);
+    const envVarUri: vscode.Uri = vscode.Uri.file(envVarPath);
+
+    await node.runWithTemporaryDescription('Downloading...', async () => {
+        ext.outputChannel.appendLine(`Downloading settings from "${client.fullName}"...`);
+        const localEnvVariables: DotenvParseOutput = await getLocalEnvironmentVariables(envVarPath, true /* allowOverwrite */);
+        const remoteEnvVariables: WebSiteManagementModels.StringDictionary = await client.listApplicationSettings();
+        if (remoteEnvVariables.properties) {
+            await confirmOverwriteSettings(remoteEnvVariables.properties, localEnvVariables, envFileName);
+        }
+
+        await fse.ensureFile(envVarPath);
+        await fse.writeFile(envVarPath, convertAppSettingsToEnvVariables(localEnvVariables, client.fullName));
+    });
+
+    const doc: vscode.TextDocument = await vscode.workspace.openTextDocument(envVarUri);
+    await vscode.window.showTextDocument(doc);
+}
+
+export function convertAppSettingsToEnvVariables(appSettings: { [propertyName: string]: string } | undefined, appName: string): string {
+    let envData: string = `# Imported Application Settings from ${appName}${os.EOL}`;
+    if (appSettings) {
+        for (const property of Object.keys(appSettings)) {
+            envData += `${property}="${appSettings[property]}"`;
```

Comment:
Will this work if the value has quotes in it?

Reply:
Seems okay!  These are the tests I tried.

![image](https://user-images.githubusercontent.com/5290572/49907103-90b9a580-fe29-11e8-96ab-1d3a7e38a8b6.png)
![image](https://user-images.githubusercontent.com/5290572/49907113-9a430d80-fe29-11e8-931a-cbe3f0d5d483.png)